Store the GH users in one secret

estib-vega opened the pull request:
- Don’t use the settings to track the authenticated users
- Use only one secret to store the users authenticated to the application

<!-- GitButler Footer Boundary Top -->
---
This is **part 1 of 1 in a stack** made with GitButler:
- <kbd>&nbsp;1&nbsp;</kbd> #10679 👈 
<!-- GitButler Footer Boundary Bottom -->

estib-vega committed:
Don't store the known users in the application settings

Remove the tracking of the known forge users from the application settings

estib-vega committed:
Use the new endpoint to fetch the GitHub usernames

Replace the settings store with the new endpoint for fetching the usernames loggin to GitHub in the application

estib-vega committed:
Add an enpoint to fetch the list of GitHub usernames

List the GitHub usernames that authenticate with the application from an endpoint

Byron commented on crates/but-github/src/token.rs:
```diff
@@ -0,0 +1,75 @@
+struct UserToken {
+    pub username: String,
+    pub access_token: String,
+}
```
Can we differentiate the token type by looking at the `access_token`? I am asking because of the semantic difference between device tokens and personal access tokens.

Byron commented on crates/but-github/src/token.rs:
```diff
@@ -0,0 +1,75 @@
+    static FAIR_QUEUE: Mutex<()> = Mutex::new(());
+    let _one_at_a_time_to_prevent_races = FAIR_QUEUE.lock().unwrap();
+    if let Some(serialized_map) =
+        secret::retrieve(GITHUB_OAUTH_ACCESS_TOKEN_MAP_KEY, secret::Namespace::Global)?
```
Since the access is now `Global`, wouldn't we run into this macOS26 bug of nightly not being able to access stable and vice versa?
Or is this now working, somehow?

Byron reviewed (commented):
Before I go, I also wanted to mention that there is this hack which completely changes what secrets we use (and how we store them) in dev mode.

https://github.com/gitbutlerapp/gitbutler/blob/9c5a1d540dafde37da018ae1504abe5ca01103fa/crates/gitbutler-tauri/src/main.rs#L103-L105

So to test this, you'd always have to do a release build, something that now at least is much faster than before.

estib-vega committed:
Persist the users in under the same secret

Instead of storing multiple secrets per GH account, store the whole list in the same place
